app/Main.hs

```diff
@@ -412,23 +412,33 @@ checkForUpdates applyMinorUpdates applyMajorUpdates = do
     isMinorReleaseFrom (x : xs) (y : ys) = y == x && ys > xs
     isMinorReleaseFrom _        _        = False
 
-verifyPackageSet :: IO ()
-verifyPackageSet = do
+verifyPackage :: Maybe String -> IO ()
+verifyPackage (Just name) = case mkPackageName (pack name) of
+  Right packageName -> verifyPackageSet (pure packageName)
+  Left pnError -> echoT . pack $ "Error while parsing input package name: " <> show pnError
+verifyPackage Nothing = echoT "A package name is required for verify"
+
+verifyPackageSet :: Maybe PackageName -> IO ()
+verifyPackageSet filterName = do
   pkg <- readPackageFile
   db <- readPackageSet pkg
+  let filteredDb = maybe id (flip Map.restrictKeys . Set.singleton) filterName $ db
 
-  echoT ("Verifying " <> pack (show (Map.size db)) <> " packages.")
-  echoT "Warning: this could take some time!"
+  case (Map.null filteredDb, filterName) of
+    (True, Just _) -> echoT . pack $ "No packages found with the name " <> show (runPackageName $ fromJust filterName)
+    _ -> do
+      echoT ("Verifying " <> pack (show (Map.size filteredDb)) <> " packages.")
+      echoT "Warning: this could take some time!"
 
-  let installOrUpdate' (name, pkgInfo) = (name, ) <$> installOrUpdate (set pkg) name pkgInfo
-  paths <- Map.fromList <$> traverse installOrUpdate' (Map.toList db)
+      let installOrUpdate' (name, pkgInfo) = (name, ) <$> installOrUpdate (set pkg) name pkgInfo
+      paths <- Map.fromList <$> traverse installOrUpdate' (Map.toList db)
 
-  for_ (Map.toList db) $ \(name, _) -> do
-    let dirFor pkgName = fromMaybe (error ("verifyPackageSet: no directory for " <> show pkgName)) (Map.lookup pkgName paths)
-    echoT ("Verifying package " <> runPackageName name)
-    dependencies <- map fst <$> getTransitiveDeps db [name]
-    let srcGlobs = map (pathToTextUnsafe . (</> ("src" </> "**" </> "*.purs")) . dirFor) dependencies
-    procs "purs" ("compile" : srcGlobs) empty
+      for_ (Map.toList filteredDb) $ \(name, _) -> do
```

Comment:
Instead of filtering the db structure, why not just extract this into a function and call it with the name of the one package you want?

Reply:
I thought about maybe making `verify` take multiple arguments but it would be a mess to start building it. I'll have to put this inner function out to a separate root definition and then revert the changes to this function otherwise

Reply:
You can just define it using `let` or `where`.